torchtext/datasets/squad1.py

```diff
@@ -29,6 +33,5 @@
 @_create_dataset_directory(dataset_name=DATASET_NAME)
 @_wrap_split_argument(('train', 'dev'))
 def SQuAD1(root, split):
-    extracted_files = download_from_url(URL[split], root=root, hash_value=MD5[split], hash_type='md5')
-    return _RawTextIterableDataset(DATASET_NAME, NUM_LINES[split],
-                                   _create_data_from_json(extracted_files))
+    saver_dp = HttpReader([URL[split]]).map(lambda x: (x[0], x[1].read())).save_to_disk(filepath_fn=lambda x: os.path.join(root, os.path.basename(x)))
+    return ParseSQuADQAData(LoadFilesFromDisk(saver_dp).parse_json_files())
```

Comment:
@VitalyFedyunin @ejguan  Can we do better? not sure if can use .map in some elegant way instead of ParseSQuADQAData pipe?

Reply:
If you use `cache_on_disk`, you can eliminate `map` IMO.

Reply:
Looks like ParseSQuADQAData is required here, it is a perfect example of the case where users need the ability to write own DataPipes to add custom processing logic.